pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/BitSlicedRangeIndexReader.java

```diff
@@ -122,35 +157,39 @@ public ImmutableRoaringBitmap getPartiallyMatchingDocIds(double min, double max)
   }
 
   private ImmutableRoaringBitmap queryRangeBitmap(long min, long max, long columnMax) {
-    RangeBitmap rangeBitmap = mapRangeBitmap();
-    if (Long.compareUnsigned(max, columnMax) < 0) {
-      if (Long.compareUnsigned(min, 0) > 0) {
-        return rangeBitmap.between(min, max).toMutableRoaringBitmap();
-      }
-      return rangeBitmap.lte(max).toMutableRoaringBitmap();
-    } else {
-      if (Long.compareUnsigned(min, 0) > 0) {
-        return rangeBitmap.gte(min).toMutableRoaringBitmap();
-      }
+    // Compare unsigned
+    boolean lowerUnbounded = min + Long.MIN_VALUE <= Long.MIN_VALUE;
+    boolean upperUnbounded = max + Long.MIN_VALUE >= columnMax + Long.MIN_VALUE;
+    if (lowerUnbounded && upperUnbounded) {
       MutableRoaringBitmap all = new MutableRoaringBitmap();
-      all.add(0, _numDocs);
+      all.add(0L, _numDocs);
       return all;
     }
+    RangeBitmap rangeBitmap = mapRangeBitmap();
+    if (lowerUnbounded) {
+      return rangeBitmap.lte(max).toMutableRoaringBitmap();
+    }
+    if (upperUnbounded) {
+      return rangeBitmap.gte(min).toMutableRoaringBitmap();
+    }
+    return rangeBitmap.between(min, max).toMutableRoaringBitmap();
```

Comment:
These are non-functional changes and I don't think they are improvements, I suggest removing them to remove noise in the PR (which is an important bug-fix)

Reply:
It will improve the case of full match, where we don't need to read the range bitmap. But agree we can focus on just the bugfix in this PR

Reply:
The data structure already detects these cases and skips them, the mapping should take ~10ns and eliminating that should be balanced against needing to read the code